cpp/src/unary/math_ops.cu

```diff
@@ -209,9 +209,31 @@ struct DeviceRInt {
   }
 };
 
-// bitwise op
+struct DeviceBitCount {
+  template <typename T>
+  std::enable_if_t<std::is_same_v<T, bool>, int32_t> __device__ operator()(T data)
+  {
+    return data ? 1 : 0;
+  }
+
+  template <typename T>
+  std::enable_if_t<!std::is_same_v<T, bool>, int32_t> __device__ operator()(T data)
+  {
+    // Convert to unsigned of the same width to handle negative numbers correctly as right shift
+    // on unsigned types is always zero fill.
+    using UnsignedT = std::make_unsigned_t<T>;
+    auto x          = static_cast<UnsignedT>(data);
+
+    int32_t count = 0;
+    while (x) {
+      count += x & 1;
+      x >>= 1;
+    }
+    return count;
```

Comment:
we should be able to use `__popc`/`__popcll` here

Reply:
Oh wow, I didn't know that.